src/app/backend/handler/apihandler.go

```diff
@@ -2264,6 +2276,44 @@ func (apiHandler *APIHandler) handleLogFile(request *restful.Request, response *
 	handleDownload(response, logStream)
 }
 
+func (apiHandler *APIHandler) handlePluginList(request *restful.Request, response *restful.Response) {
+	pluginClient, err := apiHandler.cManager.PluginClient(request)
+	if err != nil {
+		errors.HandleInternalError(response, err)
+		return
+	}
+	namespace := request.PathParameter("namespace")
+
+	result, err := plugin.GetPluginList(pluginClient, namespace)
+	if err != nil {
+		errors.HandleInternalError(response, err)
+		return
+	}
+	response.WriteHeaderAndEntity(http.StatusOK, result)
+}
+
+func (apiHandler *APIHandler) servePluginSource(request *restful.Request, response *restful.Response) {
+	pluginClient, err := apiHandler.cManager.PluginClient(request)
+	if err != nil {
+		errors.HandleInternalError(response, err)
+		return
+	}
+	k8sClient, err := apiHandler.cManager.Client(request)
+	if err != nil {
+		errors.HandleInternalError(response, err)
+		return
+	}
+	namespace := request.PathParameter("namespace")
+	pluginName := request.PathParameter("pluginName")
+
+	result, err := plugin.GetPluginSource(pluginClient, k8sClient, namespace, strings.Split(pluginName, ".")[0])
+	if err != nil {
+		errors.HandleInternalError(response, err)
+		return
+	}
+	_, _ = response.Write(result)
```

Comment:
Ignoring everything here? What if an error occurs, will it be silenced?

Reply:
Yes, it'll be silenced. I can follow the original signature if that's suggested.

Reply:
Left the error unhandled.